### DISCUSSION TITLE
Route segments without a leading dash - to support routes like /@user_one /@user_two?

### DISCUSSION THREAD

**Discussion by maxcan** [TanStack/router#1652] [Q&A]
I'm trying to migrate from React Router for a fairly large project because TanStack Router has been such a pleasure in other projects.

But, I need to support routes with a leading '@' (similar to tik tok or instagram a couple of years ago).  I can't quite figure out how to do this in TS Router.  Is it possible?

Sample routes:

/@bob
/@alice
/@etc..

but then support "conventional" routes like:

/events
/event/$eventId

Is this possible? What would be the best approach?

**Comment by etc** [comment]
Don't tag me bro.

**Reply by maxcan** [reply]
Sorry about that 😆.. completely unintentional

**Comment by SeanCassiere** [comment]
Not really.

You could handle this yourself using the `parseParams` and `stringifyParams` config options in [RouteOptions](https://tanstack.com/router/latest/docs/framework/react/api/router/RouteOptionsType#parseparams-method).

Or you could parse it in the `beforeLoad` callback and pass it down to your child routes using Route Context. Like this:

```tsx
// src/routes/$userId.tsx
export const Route = createFileRoute('/$userId')({
  beforeLoad: ({ params: { userId } }) => {
    return { userId: customParseFunction(userId) }
  },
  loader: ({ context: { userId } }) => loadUser(userId)
})

// src/routes/$userId.foo.tsx
export const Route = createFileRoute('/$userId/foo')({
  loader: ({ context: { userId } }) => loadFooDataForUser(userId)
})
```

**Reply by maxcan** [reply]
Thank you.  Parsing out the '@' isn't the problem but in this case, the `src/routes/$userId.tsx` would effectively be a wildcard route matching anything, correct?

**Reply by SeanCassiere** [reply]
Yea, more or less, provided a value is present.
